service.go

```diff
@@ -553,6 +563,39 @@ func (client *Client) ListServicesWithFramework(framework string, variables *Pay
 	return &q.Account.Services, nil
 }
 
+func (client *Client) ListServicesWithInputFilter(filters []ServiceFilterInput, variables *PayloadVariables) (*ServiceConnection, error) {
```

Comment:
```suggestion
func (client *Client) ListServicesWithInputFilter(filters ServiceFilterInput, variables *PayloadVariables) (*ServiceConnection, error) {
```
I think this should actually only take this singular input.  Because if you look at the schema for `ServiceFilterInput` if you need multiple predicates you use the predicates field on `ServiceFilterInput`. (I get that its weird that predicates also use `ServiceFilterInput` but thats how the API is built.  I think its very confusing to take in an array of `ServiceFilterInput` (even though i know the API does this) its very weird from a customer perspective because if you use multiple predicates you need to set a connective of And or Or.

Opslevel-go doesn't have to 100% mirror the API spec - we've done dozens of "slight tweaks" to make the API feel more usable.  I'm open to pushback though.

Reply:
You're totally right, good catch. The array input is because we have another format that allows a connective at the top level with filter as an array. Functionally, this is just a worse version of nesting predicates/connectives so I didn't make that structure public (or bring it to opslevel-go).